### DISCUSSION TITLE
How to export/extract data folder from container to use in a non-docker setup?

### DISCUSSION THREAD

**Discussion by Nyankoo** [felddy/foundryvtt-docker#687] [Q&A] [answered]
I'm currently looking for a way to export/extract the data folder of my docker container, so I can import it on another FoundryVTT setup that doesn't run with docker.

Is this possible?

**Comment by felddy** [comment]
Yes, the data used by FoundryVTT in the Docker container is exportable. 

How you export the data largely depends on the type of mount you used for your `/data` directory in Docker.

1. If you used a bind mount to the host OS for your `/data` directory, then the data already resides directly on your host's filesystem. In this case, you can directly copy the data to another location or even to another system.

2. If you used a Docker volume for your `/data` directory, you'll need to use Docker's `cp` command to extract the files from the `/data` directory **inside the running container** to a location on your host's filesystem.

For instance, if you'd like to copy everything from the container's `/data` directory to the `/tmp/foo` directory on your host system, you can run the following command:

```bash
docker cp container-name:/data /tmp/foo
```

Replace `container-name` with the name of your running Docker container. 

After running this command, you'll find a copy of all FoundryVTT runtime data (`Config/`, `Data/`, `Logs/` directories) in `/tmp/foo` on your host system. You'll also find the container's cache of FoundryVTT releases (`container_cache/` directory) there.

You can then use these extracted files for setting up FoundryVTT on another system.

Remember to replace `/tmp/foo` with the actual directory on your host system where you want to export the data to. Make sure this directory exists and has the appropriate permissions.

**Reply by felddy** [reply, marked as answer]
(@Nyankoo, I noticed you asked a question in a new "answer". GitHub's discussion interface can be a bit confusing, so I'm responding to your question here.)

> @felddy Thank you for the quick answer!
>
> I'm really new to Docker and when I created the container, I left the Volume Mapping empty (besides the /data), so I think it would be number 2 of what you described above, correct?

Yes, you're correct. If you didn't specify a bind mount or named volume for the `/data` directory when creating the container, Docker automatically created an anonymous volume.

If the container is still running, you can use the `docker cp` command to copy the data from the `/data` directory in the container to your host system. 

If the container is no longer running, you can still access the data in the anonymous volume. You'll need to attach this volume to a new, temporary container, then use the `docker cp` command to copy the data to your host system.
